tests/test_idpi/test_grib_decoder.py

```diff
@@ -0,0 +1,27 @@
+# Standard library
+from unittest.mock import Mock
+from unittest.mock import patch
+
+# Third-party
+import pytest
+
+# First-party
+from idpi.grib_decoder import GribReader
+
+earthkit = Mock()
+
+
+@patch("idpi.grib_decoder.earthkit.data.from_source")
+def test_ref_param_not_found(mock_from_source):
```

Comment:
I don't really get the purpose of this test. The grib field iterator is replaced by a mock object which has a very different behaviour from the real object.

Reply:
yes, you are right, a None would not be iterable. 
I added the explicit return_value = []. Is it more correct now? 

I agree with you that the test does not bring much value. 
We had already a task to reduce the domain size for testing of operators. 
If we create a tiny dataset 2x2 grid points, for example. We could easily test in a relatively quick way this on real data. We could also test construction of metadata and other functions. 
I can create a issue for this, I guess it would be for another PR. 
The question is should I then remove this tests?

Reply:
With the return value it should be fine. I usually spec the mocks to make sure that they don't allow undefined attribute access but that's overkill in this case.

I think you can leave the test but add a note in the ticket to remove the patch once the cropped data is available.

Reply:
done, I opened a ticket https://meteoswiss.atlassian.net/browse/APNRZ-200